### PR TITLE
Test new CI VMs

### PR DESCRIPTION
**Pull request by edsantiago**
From https://github.com/containers/automation_images/pull/324

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

**Comment by edsantiago** [comment]
@containers/podman-maintainers I might need help here. [Debian build](https://cirrus-ci.com/task/5857661397237760?logs=main#L1101) is failing:
```
Error: database graph driver "" does not match our graph driver "overlay": database configuration mismatch
```
The failing line of code is, I think, https://github.com/containers/podman/blob/d66b18f5afbbea16f88fe77eaa449ca9739698ea/contrib/cirrus/runner.sh#L238

My starting supposition is that it has to do with the preinstalled podman. The VMs from 3 weeks ago have `4.7.2+ds1-2`. Today it's `4.9.0+ds1-2`.  I don't know if something changed on *our* end between podman 4.7 and 4.9, or in the Debian packaging, or if this is a red herring. I'm going to dive into the rabbit hole now, am posting this in hopes that one of you says "oh yeah, that's a known bug, here's the fix"

**Comment by edsantiago** [comment]
Update: podman on debian is 100% broken. Cirrus rerun-in-terminal:
```console
# podman info
host:
  arch: amd64
  buildahVersion: 1.33.3
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon_2.1.6+ds1-1_amd64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.6, commit: unknown'
  cpuUtilization:
    idlePercent: 59.38
    systemPercent: 8.85
    userPercent: 31.77
  cpus: 2
  databaseBackend: sqlite
  distribution:
    codename: trixie
    distribution: debian
    version: "13.13"
  eventLogger: journald
  freeLocks: 2048
  hostname: cirrus-task-6317805503512576
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.6.13-cloud-amd64
  linkmode: dynamic
  logDriver: journald
  memFree: 3235119104
  memTotal: 4114952192
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns_1.4.0-5_amd64
      path: /usr/lib/podman/aardvark-dns
      version: aardvark-dns 1.4.0
    package: netavark_1.4.0-4_amd64
    path: /usr/lib/podman/netavark
    version: netavark 1.4.0
  ociRuntime:
    name: crun
    package: crun_1.14-1_amd64
    path: /usr/bin/crun
    package: crun_1.14-1_amd64
    path: /usr/bin/crun
    version: |-
      crun version 1.14
      commit: 667e6ebd4e2442d39512e63215e79d693d0780aa
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt_0.0~git20231230.f091893-1_amd64
    version: |
      pasta unknown version
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: true
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    package: crun_1.14-1_amd64
    path: /usr/bin/crun
    version: |-
      crun version 1.14
      commit: 667e6ebd4e2442d39512e63215e79d693d0780aa
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt_0.0~git20231230.f091893-1_amd64
    version: |
      pasta unknown version
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: true
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns_1.2.1-1_amd64
    version: |-
      slirp4netns version 1.2.1
      commit: 09e31e92fa3d2a1d3ca261adaeb012c8d75a8194
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.5
  swapFree: 0
  swapTotal: 0
  uptime: 0h 1m 15.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan  - ipvlan
  volume:
  - local
registries: {}
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 211116445696
  graphRootUsed: 3995324416
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.9.0
  Built: 0
  BuiltTime: Thu Jan  1 00:00:00 1970
  GitCommit: ""
  GoVersion: go1.21.6
  Os: linux
  OsArch: linux/amd64
  Version: 4.9.0
```
That was the first one. It took about 5-10 seconds. Immediately afterward:
```console
# time podman info
Error: database graph driver "" does not match our graph driver "overlay": database configuration mismatch

real    0m0.068s
user    0m0.015s
sys     0m0.026s
root@cirrus-task-6317805503512576:/tmp/.cirrus# podman system reset
Error: database graph driver "" does not match our graph driver "overlay": database configuration mismatch
root@cirrus-task-6317805503512576:/tmp/.cirrus# podman --graph-driver overlay system reset
Error: unknown flag: --graph-driver
root@cirrus-task-6317805503512576:/tmp/.cirrus# podman info
Error: database graph driver "" does not match our graph driver "overlay": database configuration mismatch
root@cirrus-task-6317805503512576:/tmp/.cirrus# 
```

**Comment by Luap99** [comment]
It is likely that this is on out end, debian does not ship a default storage driver AFAIK and somehow the db gets no driver the first time we write the storage driver into it but after that the code gets the driver and complains cc @mheon 

I guess one way to try to reproduce is to make sure storage.conf does not contain a driver (empty). I have no tried that.

**Comment by edsantiago** [comment]
`/etc/containers/storage.conf` is written by us in CI:
```
[storage]
driver = "overlay"
runroot = "/run/containers/storage"
graphroot = "/var/lib/containers/storage"
```
This is part of the system-test work I did last year to fix the bug where we _thought_ we were testing X but were testing Y instead.

I think `rm -rf /var/lib/containers` is making podman work again. I will try that in my next push.

**Comment by Luap99** [comment]
> `/etc/containers/storage.conf` is written by us in CI:
> 
> ```
> [storage]
> driver = "overlay"
> runroot = "/run/containers/storage"
> graphroot = "/var/lib/containers/storage"
> ```
> 
> This is part of the system-test work I did last year to fix the bug where we _thought_ we were testing X but were testing Y instead.
> 
> I think `rm -rf /var/lib/containers` is making podman work again. I will try that in my next push.

Yes but after the first podman command maybe? That would explain why it is not happy at all because you should not change storage.conf without running podman system reset before it.

**Comment by edsantiago** [comment]
Running system-podman before built-podman would be a disaster in some circumstances (i.e. like this one): it would leave behind state that could change the behavior of built-podman.

In the past I've taken great pains to make sure that none of our setup scripts ever run podman-anything-ever. It's possible that maybe the debian postinst scripts invoke podman? Or maybe something snuck in?

Anyhow, my `rm -rf` seems to have gotten us past that crisis, and I'm tempted to leave it in (with some cleanup). I strongly prefer `rm -rf /var/lib/containers/storage` over `podman system reset` because that would require system-podman which, see above. But I realize either approach has its merits and would like to hear differing thoughts.

**Comment by Luap99** [comment]
I guess this runs before we build the new?
https://github.com/containers/podman/blob/d66b18f5afbbea16f88fe77eaa449ca9739698ea/contrib/cirrus/lib.sh#L297

**Comment by edsantiago** [comment]
OUCH! Nice catch. And it's obviously my fault: it was my express intention in #20559, but apparently I did not remove the problematic line that I had meant to delete. Thank you for catching that.

**Comment by edsantiago** [comment]
Looks like images are good! Marking draft in order to wait for  https://github.com/containers/automation_images/pull/324 which was trickier than predicted.

**Review by giuseppe — APPROVED** [review]
https://github.com/containers/automation_images/pull/324 is all green

LGTM

**Comment by rhatdan** [comment]
LGTM

**Comment by edsantiago** [comment]
Ready to go. I've seen flakes, but they seem to be quay-hiccup-related.

**Comment by edsantiago** [comment]
Update: also failed [rawhide rootless](https://api.cirrus-ci.com/v1/artifact/task/6730819423698944/html/int-podman-rawhide-rootless-host-sqlite.log.html#t--Podman-run-networking-podman-verify-network-scoped-DNS-server-and-also-verify-updating-network-dns-server--1), with aardvark 1.9 and netavark 10, suggesting that the bug is in **netavark**, not aardvark.

File: test/e2e/run_networking_test.go
Comment (edsantiago) [inline test/e2e/run_networking_test.go:27]: I will keep trying to rebuild CI VMs, and once I get a working netavark, will remove these skips.

**Review by edsantiago — COMMENTED** [review]
@containers/podman-maintainers @ashley-cui PTAL, let's get this in so we can unblock #21410.

**Comment by ashley-cui** [comment]
Thanks!

/hold
/lgtm

**Comment by edsantiago** [comment]
/hold cancel

Thanks for your patience, @ashley-cui. Please rebase your PR on this and cross fingers...

**Pull request merged**
